astropy/utils/data.py

```diff
@@ -1118,7 +1114,7 @@ def is_url_in_cache(url_key):
 
 
 def _do_download_files_in_parallel(args):
```

Comment:
Do we even still need this? Can `map` just call `download_file` directly now?

Reply:
I don't think so, it still requires the unpacking.

Reply:
Maybe there is a way to unpack args without using a dummy in-between function? Anyway, that should not block the merge.

Reply:
Well, there is `starmap` (however it's more like `map` than the currently used `imap`) and maybe something in `concurrent.futures`, however I didn't want to rewrite `ProcessBar`... and it wouldn't be a clean backport to 2.0.3 because of the Python 2 support. Also it would mix a straightforward bugfix to something like an enhancement.

Reply:
A decorator would be easy to write (but again that would make it an enhancement rather than a bugfix). Something like (shameless self-promotion ahead: [`packed`](https://iteration-utilities.readthedocs.io/en/stable/generated/packed.html)).

Reply:
hey, you don't have a link to the source repo at any prominent place?

Reply:
It's essentially just:

```
class packed(object):
  __slots__ = ('func', )
  def __init__(self, func):
    self.func = func
  def __call__(self, args, **kwargs):
    return self.func(*args, **kwargs)
```

but [written as C extension](https://github.com/MSeifert04/iteration_utilities/blob/master/src/packed.c) (with some minor additional stuff).

Reply:
(yes, I found it, I just meant that from the docs of `iteration_utilities` one has to dig down to find a link to github ;) )

Reply:
Seems like I, finally, have a reason to look into "how to create a footer/header" with sphinx. 😄

Reply:
you can opt in the annoying "edit on github" plugin, that's the easiest ;)

Reply:
I'm almost certain that it won't work because of the C extension (like the `[source]` "button").

Reply:
There is a way to attach docstring into your C-extension. See [astropy/wcs/docstrings.py](https://github.com/astropy/astropy/blob/master/astropy/wcs/docstrings.py).

Reply:
I don't doubt that, but the "edit on github" and "source" buttons always refer to the Python function or don't display.

But I'm getting side-tracked. 😄

Reply:
That is true. It does not point straight to the C code.